gnmi/gnmiclient/cache.go

```diff
@@ -19,36 +19,47 @@ import (
 	"io"
 
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/peer"
 
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/lemming/gnmi/gnmistore"
 )
 
-// NewCacheClient creates a gNMI client for the gNMI cache.
+// New creates a state-based gNMI client for the gNMI cache.
 // The client calls the server gRPC implementation with a custom streaming gRPC implementation
 // in order to bypass the regular gRPC wire marshalling/unmarshalling handling.
-// TODO: refactor gNMI so that a single server can be used here.
-func NewCacheClient(srv gpb.GNMIServer, setClient gpb.GNMIClient) gpb.GNMIClient {
+func New(srv gpb.GNMIServer) (gpb.GNMIClient, error) {
 	return &cacheClient{
-		srv:       srv,
-		setClient: setClient,
-	}
+		gnmiMode: gnmistore.StateMode,
+		srv:      srv,
+	}, nil
+}
+
+// NewReadOnly creates a config-based gNMI client for the gNMI cache.
+// The client calls the server gRPC implementation with a custom streaming gRPC implementation
+// in order to bypass the regular gRPC wire marshalling/unmarshalling handling.
+func NewReadOnly(srv gpb.GNMIServer) (gpb.GNMIClient, error) {
```

Comment:
do we need this? i don't see where it's used

Reply:
True, I don't see this being used.